lib/rate_matrix.ml

```diff
@@ -138,6 +138,30 @@ module Nucleotide = struct
         else if Nucleotide.transversion i j then 1. /. (kappa +. 2.)
         else kappa /. (kappa +. 2.)
       )
+
+  let hky85 ~equilibrium_frequencies:(stationary_distribution : Nucleotide.vector)
+      ~alpha_transition ~beta_transversion =
+    let m = make (fun i j ->
+        let coef = if Nucleotide.equal i j
+          then -1.
+          else if Nucleotide.transversion i j
+          then beta_transversion
+          else alpha_transition
+        in
+        coef *. stationary_distribution.Nucleotide.%(j)
+      ) in
+    scaled_rate_matrix stationary_distribution m
+
+  let%test "HKY85 stationary distribution" =
+    let rng = Utils.rng_of_int 420 in
+    let pi = Nucleotide.random_profile rng 10. in
+    let rate_params = Utils.random_profile rng 2 in
```

Comment:
rates have no reason to sum to one, they just need to be positive. Maybe it's better not to test on this particular case

Reply:
ac5fe5e uses rates drawn from uniform distribution, without constraint on the sum

Reply:
:ok_hand: